msteams-platform/concepts/deploy-and-publish/appsource/prepare/frequently-failed-cases.md

```diff
@@ -1,246 +1,333 @@
+
+### 2.4 External domains
+
+Tunneling services and domains outside of an organization’s control, including wildcards, must not be included in the valid domains of the app manifest.
+
+### 2.5 App permissions
+
+Apps that request permissions during the Azure Active Directory (Azure AD) consent process must provide business justification for these requests.
+
+### 2.6 Authentication
+
+#### 2.6.1 Signing out
+
+The ability to sign out from any authenticated external services must be present within the app. For more information, see [authentication in Teams](~/concepts/authentication/authentication.md).
+
+When a user logs out, they must log out only from the app and not from the Teams client.
+
+#### 2.6.2 Unsharing content
+
+Teams apps that require authentication with an external service to share content in channels must clearly state in help documentation (or similar resources) how to disconnect or unshare content if that feature is supported on the external service. This does not mean the ability to unshare content must be present in your Teams app.
+
+#### 2.6.3 Government Community Cloud listings
+
+To make your app available for Government Community Cloud (GCC) users and to avoid duplicate app listings in the store, the authentication process must identify and route GCC users to the specified or expected content URL.
+
+### 2.7 Health information sources
+
+Apps catering to a patient’s health records, diseases, wellness, nutrition, or pandemic-related information must abide with the user's local health laws. In addition, the information provided by such apps must always be from trusted sources such as the World Health Organization (WHO) or a government website.
+
+### 2.8 Sensitive content
+
+Your app must not post sensitive data, such as credit card or financial payment instrument data. The app also must not display health, contact tracing, or other personally identifiable information (PII) to an audience not intended to view that content.
+
+Warn users before your app downloads any files or executables (.exe) into the user's machine or environment.
+
+## 3.0 General functionality and performance
+
+### 3.1 Launching external functionality
+
+Apps must not take the user out of Teams for core user scenarios. Task modules, cards or tabs are recommended to display information to the user within Teams. Link targets in apps must not link to an external browser but must link to `div` elements contained within Teams, for example, inside task modules and tabs.
+
+For scenarios that require external functionality, your app must have explicit user permission to launch functionality outside of Teams.
+
+### 3.2 Compatibility
+
+Apps must be fully functional on the following operating systems and browsers:
+
+* Microsoft Windows 7 and later
+* macOS 10.10 and later
+* Microsoft Edge 12 and later
+* Internet Explorer 11 and later
+* Mozilla Firefox 47.0 and later
+* Google Chrome 51.0 and later
```

Comment:
Update to :

- Microsoft Windows 7 and later
- macOS 10.10 and later
- Microsoft Edge 12 and later
- Mozilla Firefox 47.0 and later
- Google Chrome 51.0 and later
- iOS 9.0 and later
- Android 4.4 and later